internal/messagequeue/messagequeue.go

```diff
@@ -740,13 +740,6 @@ func (mq *MessageQueue) extractOutgoingMessage(supportsHave bool) (bsmsg.BitSwap
 
 	// Next, add the wants. If we have too many entries to fit into a single
 	// message, sort by priority and include the high priority ones first.
-	// However, avoid sorting till we really need to as this code is a
-	// called frequently.
-
-	// Add each regular want-have / want-block to the message.
-	if msgSize+(len(peerEntries)*bsmsg.MaxEntrySize) > mq.maxMessageSize {
```

Comment:
We're now _eagerly_ sorting which could affect performance. However, I believe the _real_ perf issue was that we were repeatedly sorting the same broadcast list over-and-over, which we're now doing at most once per change.